src/NzbDrone.Core/Notifications/Trakt/TraktService.cs

```diff
@@ -0,0 +1,261 @@
+            }
+        }
+
+        public void AddEpisodeToCollection(TraktSettings settings, Series series, EpisodeFile episodeFile)
+        {
+            var payload = new TraktCollectShowsResource
+            {
+                Shows = new List<TraktCollectShow>()
+            };
+
+            var traktResolution = MapResolution(episodeFile.Quality.Quality.Resolution, episodeFile.MediaInfo?.ScanType);
+            var mediaType = MapMediaType(episodeFile.Quality.Quality.Source);
+            var audio = MapAudio(episodeFile);
+            var audioChannels = MapAudioChannels(episodeFile, audio);
+
+            var payloadEpisodes = new List<TraktEpisodeResource>();
+
+            foreach (var episode in episodeFile.Episodes.Value)
+            {
+                payloadEpisodes.Add(new TraktEpisodeResource
+                {
+                    Number = episode.EpisodeNumber,
+                    CollectedAt = DateTime.Now,
+                    Resolution = traktResolution,
+                    MediaType = mediaType,
+                    AudioChannels = audioChannels,
+                    Audio = audio,
+                });
+            }
+            
+            var payloadSeasons = new List<TraktSeasonResource>();
+            payloadSeasons.Add(new TraktSeasonResource
+            {
+                Number = episodeFile.SeasonNumber,
+                Episodes = payloadEpisodes
+            });
+
+            payload.Shows.Add(new TraktCollectShow
+            {
+                Title = series.Title,
+                Year = series.Year,
+                Ids = new TraktShowIdsResource
+                {
+                    Tvdb = series.TvdbId,
+                    Imdb = series.ImdbId ?? "",
+                },
+                Seasons = payloadSeasons,
+            }); ;
+
+            _proxy.AddToCollection(payload, settings.AccessToken);
+        }
+
+        private string MapMediaType(QualitySource source)
+        {
+            var traktSource = string.Empty;
+
+            switch (source)
+            {
+                case QualitySource.Web:
+                case QualitySource.WebRip:
+                    traktSource = "digital";
+                    break;
+                case QualitySource.BlurayRaw:
+                case QualitySource.Bluray:
+                    traktSource = "bluray";
+                    break;
+                case QualitySource.Television:
+                case QualitySource.TelevisionRaw:
+                    traktSource = "vhs";
+                    break;
+                case QualitySource.DVD:
+                    traktSource = "dvd";
+                    break;
+            }
+
+            return traktSource;
+        }
+
+        private string MapResolution(int resolution, string scanType)
+        {
+            var traktResolution = string.Empty;
+            var interlacedTypes = new string[] { "Interlaced", "MBAFF", "PAFF" };
```

Comment:
This doesn't need to be recreated each time and would be better as a HashSet, similar to how it's done for subtitle file extensions: https://github.com/Sonarr/Sonarr/blob/0a2b109a3fe101e260b623d0768240ef8b7a47ae/src/NzbDrone.Core/Extras/Subtitles/SubtitleFileExtensions.cs

Reply:
I'll work on fixing this up soon...